snowflake/snowflake.go

```diff
@@ -168,7 +168,10 @@ func (g *Generator) setNodeIDProvider(nodeIDProvider NodeIDProvider) (err error)
 // Decompose returns a map of snowflake id parts.
 func (g *Generator) Decompose(sid uint64) (id ID) {
 	timestamp := sid >> timeShift
-	id.Timestamp = time.UnixMilli(g.Epoch.UnixMilli() + int64(timestamp*timeUnit))
+	// FIXME:
+	// after updating to go 1.17 can be replaced with
```

Comment:
@mysqto
So this `func` is not available in `1.16`?

Reply:
Yes @viettd   Ref: https://golang.org/doc/go1.17

Reply:
I see they just release `1.17.1` a few days back (fixed a few issues found on `1.17.0`), shall we do the upgrade for product repos?

Reply:
I think we can keep using go 1.16, and use this impl, if somebody also using this repo and not updated to go `1.17`, the code can not be complied

Reply:
Ok then.